.github/workflows/main-cmake.yml

```diff
@@ -135,10 +135,6 @@ jobs:
 
       # CMake gets libaec from fetchcontent
 
-      - name: Install Dependencies (Windows)
```

Comment:
Maybe - Ninja will be needed if we want to use clang (like OneAPI, it is needed for alternative compilers on windows).  Note that ninja is required for the binary tests.

Reply:
I think we need to be cautious about what checks are required for PRs if they're going to go out and grab things from the internet outside of what's the default on runners. Testing alternative compilers on Windows is great, but probably doesn't need to be a required check.

Reply:
Clang and OneAPI on windows are just as important as those compilers for linusx, right?
Unfortunately GH actions for testing are problematic at any time and chocaoately isn't necessarily worse then anything else being used.
The more important metric is time for PRs S.B. less than 25 mins for a workflow.

Reply:
Maybe main-cmake.yml would be better named main-cmake-vs.yml or vs-cmake.yml?

Reply:
At the very least we should be able to keep a cached installation of ninja and other required things around when we've gotten them once. It also doesn't seem like Ninja is used anywhere for the binary tests. It seems like just the OneAPI Windows action is using it.

Reply:
We may also look at https://cmake.org/cmake/help/latest/variable/CMAKE_GENERATOR_TOOLSET.html. Seems like we can use clang and possibly OneAPI as well with VS generators so that we don't need ninja in the first place.

Reply:
bintest definitely uses ninja as the generator.